include/drivers/flash.h

```diff
@@ -254,16 +251,17 @@ static inline int z_impl_flash_erase(const struct device *dev, off_t offset,
  *
  *  @return  0 on success, negative errno code on fail.
  */
+__deprecated
 __syscall int flash_write_protection_set(const struct device *dev,
 					 bool enable);
 
 static inline int z_impl_flash_write_protection_set(const struct device *dev,
 						    bool enable)
 {
-	const struct flash_driver_api *api =
-		(const struct flash_driver_api *)dev->api;
+	ARG_UNUSED(dev);
+	ARG_UNUSED(enable);
 
-	return api->write_protection(dev, enable);
+	return 0;
```

Comment:
It strikes me that making this function a no-op will break out-of-tree flash drivers that do not yet incorporate automatic write protection management, which isn't quite the intent behind deprecation.

Reply:
I'm not sure whether zephyr should thread drivers shim implementation API similar as driver user API.
For my point of view `api->write_protection` is something internal (so I don't care out-off-tree driver).
@carlescufi Any thought on that?

Reply:
Will ask @anangl I bet he has an experience on such things.